### PR TITLE
Add User Domain Blocks

### PR DESCRIPTION
**Pull request by dansup**
Adds the ability for users to block specific domains by preventing interactions and content from the domain from appearing in feeds.

### Add domain block for all users

To add a domain block for all users while allowing them to optionally remove it, run the following command:

```
php artisan app:add-user-domain-block
```

### Remove domain block for all users

To remove a domain block for all users while allowing them to optionally re-add it, run the following command:

```
php artisan app:delete-user-domain-block
```

### Fetch blocked domains
- GET `/api/v1/domain_blocks` - Returns Array of Domains, [like Mastodon](https://docs.joinmastodon.org/methods/domain_blocks/#get)

### Add blocked domain
- POST `/api/v1/domain_blocks` - With `domain` param, [like Mastodon](https://docs.joinmastodon.org/methods/domain_blocks/#block)

### Remove blocked domain
- DELETE `/api/v1/domain_blocks` - With `domain` param, [like Mastodon](https://docs.joinmastodon.org/methods/domain_blocks/#unblock)

### Pagination

We use `Link` header pagination similar to Mastodon, however we use cursor based pagination and do not support `min_id`, `max_id` or `since_id` parameters.

#### Example Link Header Pagination (with next only)

```
Content-Encoding: gzip
Content-Type: application/json
Date: Sat, 16 Dec 2023 04:20:69 GMT
Link: <https://pixelfed.test/api/v1/domain_blocks?cursor=eyJpZCI6NDY3LCJfcG9pbnRzVG9OZXh0SXRlbXMiOnRydWV9&limit=200>; rel="next"
Server: nginx
```

#### Example Link Header Pagination (with next + prev)

```
Content-Encoding: gzip
Content-Type: application/json
Date: Sat, 16 Dec 2023 04:20:69 GMT
Link: <https://pixelfed.test/api/v1/domain_blocks?cursor=eyJpZCI6NTM4LCJfcG9pbnRzVG9OZXh0SXRlbXMiOnRydWV9&limit=200>; rel="next", <https://pixelfed.test/api/v1/domain_blocks?cursor=eyJpZCI6NzM3LCJfcG9pbnRzVG9OZXh0SXRlbXMiOmZhbHNlfQ&limit=200>; rel="prev"
Server: nginx
```

## Todo

- [x] Model
- [x] Migration
- [x] Config
- [x] Api Endpoints
- [x] Federation
- [x] HomeFeedFanout
- [x] Public/Network feed filtering
- [x] Hashtag feed filtering
- [x] Search results filtering
- [x] Add Domain Block Admin command
- [x] Remove Domain Block Admin command

### CODE REVIEW

File: config/instance.php [hunk 1]
```diff
@@ -110,7 +110,8 @@
 
 	'user_filters' => [
 		'max_user_blocks' => env('PF_MAX_USER_BLOCKS', 50),
-		'max_user_mutes' => env('PF_MAX_USER_MUTES', 50)
+		'max_user_mutes' => env('PF_MAX_USER_MUTES', 50),
+		'max_domain_blocks' => env('PF_MAX_DOMAIN_BLOCKS', 50),
```
Comment (ThisIsMissEm) [inline config/instance.php:114]: This is hardcoded elsewhere to 400 I think?

File: resources/views/settings/privacy.blade.php [hunk 2]
```diff
@@ -8,8 +8,9 @@
   <hr>
   <div class="form-group pb-1">
     <p>
-      <a class="btn btn-outline-secondary py-0 font-weight-bold" href="{{route('settings.privacy.muted-users')}}">Muted Users</a>
-      <a class="btn btn-outline-secondary py-0 font-weight-bold" href="{{route('settings.privacy.blocked-users')}}">Blocked Users</a>
+      <a class="btn btn-link py-0 font-weight-bold" href="{{route('settings.privacy.muted-users')}}">Muted Accounts</a>
+      <a class="btn btn-link py-0 font-weight-bold" href="{{route('settings.privacy.blocked-users')}}">Blocked Accounts</a>
+      <a class="btn btn-link py-0 font-weight-bold" href="{{route('settings.privacy.domain-blocks')}}">Domain Blocks</a>
```
Comment (ThisIsMissEm) [inline resources/views/settings/privacy.blade.php]: Perhaps: "Blocked Domains"?

File: config/instance.php (hunk 1 above)
Comment (dansup) [reply config/instance.php:114]: Nope, did you mean this?

https://github.com/pixelfed/pixelfed/blob/60e053c93671d0e790cde02aa1c51bc011cca17a/app/Http/Controllers/Api/V1/DomainBlockController.php#L70

The `400` there is the http status code to be returned

File: resources/views/settings/privacy.blade.php (hunk 2 above)
Comment (dansup) [reply resources/views/settings/privacy.blade.php]: Good call, done!

File: config/instance.php (hunk 1 above)
Comment (ThisIsMissEm) [reply config/instance.php:114]: Oooh! Ha, that's what I get for reviewing on mobile when sleepy

**Review by x64x2 — APPROVED** [review]

**Pull request merged**